cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -1153,7 +1153,10 @@ class TableSorter {
 // ----------------------------------------------------------------------
 // Top-level sort functions
 
-const auto kDefaultSortOptions = SortOptions::Defaults();
+static const ArraySortOptions* GetDefaultArraySortOptions() {
+  static const auto kDefaultArraySortOptions = ArraySortOptions::Defaults();
```

Comment:
1) You're getting a redeclared identifier because of `GetDefaultArraySortOptions` in unity builds (see e.g. AppVeyor failure)
2) This should be `SortOptions`, not `ArraySortOptions`

Reply:
Done.